app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedFragment.java

```diff
@@ -0,0 +1,216 @@
+            Navigator.openCategoryDialogFragment(this, Objects.requireNonNull(getActivity()));
+        });
+        return root;
+    }
+
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putSerializable(CURRENT_CATEGORY, presenter.getCategory());
+        outState.putBoolean(CURRENT_SHOW_FAVORITE, presenter.getShowFavorite());
+    }
+
+    @Override
+    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+        super.onViewStateRestored(savedInstanceState);
+
+        if (savedInstanceState != null) {
+            Category category = (Category) savedInstanceState.getSerializable(CURRENT_CATEGORY);
+            Boolean showFavoriteNews = savedInstanceState.getBoolean(CURRENT_SHOW_FAVORITE);
+            presenter.setShowFavorite(showFavoriteNews);
+            presenter.setCategory(category);
+        }
+
+    }
+
+    private void initRecyclerView() {
+        adapter = new NewsFeedAdapter(context);
+        recyclerView.setLayoutManager(new LinearLayoutManager(context));
+        recyclerView.setAdapter(adapter);
+        adapter.setOnItemClickListener(new NewsFeedAdapter.OnNewsItemClickListener() {
+            @Override
+            public void onChangeFavoriteStateClick(int position) {
+                presenter.changeNewsFavoriteState(position);
+            }
+
+            @Override
+            public void OnRecyclerItemClick(int position) {
+                presenter.openNewsDetails(position);
+            }
+        });
+    }
+
+    @Override
+    public void setFavoriteIcon(Boolean showOnlyFavorite) {
+        if (showOnlyFavorite) {
+            favoriteNewsButton.setImageDrawable(IconicStorage.getYellowStarBorderless(context));
+        } else {
+            favoriteNewsButton.setImageDrawable(IconicStorage.getWhiteStarBorderless(context));
+        }
+    }
+
+    @Override
+    public void setCategoryTitle(Category category) {
+        updateCategoryNameOnDescription(Category.getDisplayName(category));
+    }
+
+    @Override
+    public void updateNewsItem(int position) {
+        adapter.notifyItemChanged(position);
+    }
+
+    @Override
+    public void removeNewsItem(int position) {
+        adapter.notifyItemRemoved(position);
+    }
+
+    @Override
+    public void showProgressBar() {
+        progressBar.setVisibility(ProgressBar.VISIBLE);
+    }
+
+    @Override
+    public void hideProgressBar() {
+        progressBar.setVisibility(ProgressBar.GONE);
+
+    }
+
+    private void setEmptyViewOnDisplay() {
+        recyclerView.setVisibility(View.GONE);
+        emptyView.setVisibility(View.VISIBLE);
+    }
+
+    private void setRecyclerViewOnDisplay() {
+        recyclerView.setVisibility(View.VISIBLE);
+        emptyView.setVisibility(View.GONE);
+    }
+
+    @Override
+    public void updateNewsListUI(List<NewsItem> news) {
+        Log.d(TAG, "updateNewsListUI: ");
```

Comment:
1. I don't think we need any kind of logging here.
2. This logic is a part of the presenter. A presenter should decide what to do and call a corresponding `View` method (and pass a parameter if required).